compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -1421,6 +1421,8 @@ private void processIdentifierEnd() {
                 case LexerTerminals.NEWLINE:
                 case LexerTerminals.CARRIAGE_RETURN:
                 case LexerTerminals.TAB:
+                    reader.advance();
+                    reportInvalidEscapeSequence((char) 0);
```

Comment:
The error msg shown to the user should be  `invalid escape sequence '\'`
```suggestion
                    reportLexerError(DiagnosticErrorCode.ERROR_INVALID_ESCAPE_SEQUENCE, "");
```

Reply:
Will update the documentation lexer. Here `reportInvalidEscapeSequence((char) 0);` gives the expected error right?

Reply:
Following is from the test case we have added. 
`BAssertUtil.validateError(resultNeg, i++, "invalid escape sequence '\\\u0000'", 4, 9);`

That string is displayed as below in IntelliJ
![Screenshot from 2021-09-13 14-43-53](https://user-images.githubusercontent.com/39232462/133057806-a9dcd175-e46a-4cb7-8d77-616bbc690f1d.png)

Reply:
Will update

Reply:
Updated